### PR TITLE
Added: Payment property on Charge model

### PR DESCRIPTION
**Pull request by pitiphong-p**
The payment property hold the payment data telling that this charge is paid by card or offsite payment

### CODE REVIEW

File: OmiseSwift/Charge.swift [hunk 1]
```diff
@@ -48,6 +48,21 @@ public class Charge: ResourceObject {
...
+        set { set("offsite", OffsitePaymentConverter.self, toValue: newValue) }
+    }
+    
+    public var payment: Payment? {
+        switch get("source_of_fund", StringConverter.self) {
+        case "offsite"?:
+            return offsite.map(Payment.offsite)
+        case "card"?, nil /* nil for backward compatability */ :
+            return card.map(Payment.card)
+        default: return nil
```
Comment (chakrit) [inline OmiseSwift/Charge.swift:62]: Wait ☝️ is `nil` case and `default` case the same?

File: OmiseSwift/Payment.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+extension Payment: Equatable {
+    /// Returns a Boolean value indicating whether two values are equal.
+    ///
+    /// Equality is the inverse of inequality. For any values `a` and `b`,
+    /// `a == b` implies that `a != b` is `false`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
+    public static func ==(lhs: Payment, rhs: Payment) -> Bool {
```
Comment (chakrit) [inline OmiseSwift/Payment.swift:18]: Is it possible to do avoid doing this by having `Payment`, `OffsitePayment` and `Card` implements `==` ?

**Review by chakrit — APPROVED** [review]
Just minor issues. Overall looks ok.

File: OmiseSwift/Charge.swift (hunk 1 above)
Comment (pitiphong-p) [reply OmiseSwift/Charge.swift:62]: No, `default` is for any others string that is not `nil` in the case

File: OmiseSwift/Payment.swift (hunk 2 above)
Comment (pitiphong-p) [reply OmiseSwift/Payment.swift:18]: Well this PR includes the implantation of `==`. Not sure what you mean though

**Comment by igroomgrim** [comment]
👍

**Review by igroomgrim — APPROVED** [review]

**Pull request merged**